libraries/app/include/graphene/app/api.hpp

```diff
@@ -185,30 +185,31 @@ namespace graphene { namespace app {
                                                                         unsigned limit = 100,
                                                                         uint32_t start = 0) const;
          /**
-          * @brief Get asset pairs with b:a pair
+          * @brief Get order book with b:a pair
```

Comment:
Absolutely wrong.

Reply:
What is better in your opinion ? Because trader told me this variant with `"order book"` is better, but it isn't. Can you please assist me just a bit ?

Thanks !

Reply:
An API querying for history won't return an order book, that's why.
Please try understand the API, then explain it to the trader and get feedback. Modifying code cluelessly is a waste of time.

Sure I can give my opinion or even write the document by myself, however, since you asked to work on this issue, you should try to get it done by yourself.

Reply:
I've changed "order book" to "trading orders", is it wrong ?

Reply:
IMHO, "trading order" is a confusing phrase. Why not ask traders?

Reply:
I've asked in DEX Trading group in Telegram all traders and just 1 answer got from Jacob Walrus who told me that "order book" is better choice.
Jacob Walrus is a founder of EasyDex as he told me:
"and just so you know who I am, I'm the founder (co-founder with my wife technically) of EasyDex"

Reply:
IMHO as I thought since beginning **"fill ordrer"** name (the same as API call contains) is just exactly what we need, but tried to find more suitable keywords. So maybe **"fill order"** is the best choice, will research more this question, more deeply ...

Thanks !

Reply:
After my researches within code API and different useful Trading info in web I've found 1 new solution-keyword: **"open warrant"** instead of "trading pairs", what do you think about it ? Isn't it confusing ?
```
@brief Get open warrants with b:a asset pairs
```
I think it makes more sense and is suitable here. What are your thoughts ?

Thanks !

Reply:
I've never seen such a word elsewhere, but I guess it's wrong. Seems you're still in the wrong direction. In order to show that you've understood the API, before you write document, can you please explain what this API is, in plain English?

Reply:
> Jacob Walrus who told me that "order book" is better choice.

I guess you asked wrong question to him or in the channel, thus got a unrelated answer.

Reply:
the main idea of `get_fill_order_history` API method is to retrieve a vector (limited in size) of `fill_order_operations` with b:a asset pairs.

Reply:
Correct. Then what does a "fill_order_operation" object mean?

Reply:
The purpose of `fill_order_operation` is tracking account history, also it is a `virtual` operation.

Reply:
here are all members of `fill_order_operation`:

```
      object_id_type      order_id;
      account_id_type     account_id;
      asset               pays;
      asset               receives;
      asset               fee; // paid by receiving account
      price               fill_price;
      bool                is_maker;
```

Reply:
so `fill_order_operation` is needed when we want to place a trading order, it is required to fill the form on either the ask or the bid side (respectively, buy or sell side). You will need to define a price and an amount to sell/buy. The cost for this order will be calculated automatically.

Reply:
Check when will a `fill_order_operation` be created.

Reply:
In addition, what is a "virtual" operation.

Reply:
`virtaul operations` are those operations (derived from `base_operation` also) which are produced by blockchain itself, thus means they are not triggered by users.

Reply:
> they are not directly triggered by user operations

Not exactly.

Why the blockchain itself "produces" such operations?

Why this API is querying for such operations?

Reply:
**`fill_order_operation`** is created just in next 3 places:
1. **`fill_limit_order`** https://github.com/bitshares/bitshares-core/blob/develop/libraries/chain/db_market.cpp#L782
2. **`fill_call_order`** https://github.com/bitshares/bitshares-core/blob/develop/libraries/chain/db_market.cpp#L866
3. **`fill_settle_order`** https://github.com/bitshares/bitshares-core/blob/develop/libraries/chain/db_market.cpp#L894

Reply:
@abitmore Thank you for help ! Your questions helped me a lot to understood more details in correct way/direction. I want to research more deeply the code and especially real & virtual operations. Also I want to answer on your 2 questions:

> Why the blockchain itself "produces" such operations?
> 
> Why this API is querying for such operations?

1. For matching orders
2. For tracking account history and to speed up `replay` process

If you have any other questions, please just ask me and I will try to answer on all them.

Reply:
Both answers are not accurate, although some comments in the code said so, the comments are outdated.

Virtual operations can be seen as (a subset of) results of processing other operations or processing post-transaction chain logic. `fill_order_operation` objects are NOT "for matching orders", instead, they're results of "matching orders". Why the blockchain produced them? For easier tracking. A trader wants to know what's the result when he placed an order, so every time his order is filled (understand what is `fill`?), a `fill_order_operation` object is created, when the trade queries for his account history, he can see the results. It has nothing to do with replay speed.

However, in this issue we're talking about market history but not account history. Now can you explain it?

Reply:
According to the link in your comment:
> Matching orders is the process ...

The API doesn't return a "process" or list of "process".

Reply:
Investopedia IS correct. However, to fit our doc, you need to adjust the wording according to English grammar.

Reply:
I thought **"matching orders"** is `stable phrase` in trader's slang

Reply:
after our discussion and a lot of efforts trying to figure out the best suitable solution with @abitmore about API doc, `fill_order_operation` and after my code researches I've some results for your judgement:

1. **"matching orders"** can be filled partially, so traders will be confused with such API doc description (@abitmore helped me here with more details & examples, thanks a lot !)
2. this API method `get_fill_order_history` will return `filled orders` which also are `confirmed` by blockchain, so the best solution is to use **"confirmed orders"**, thus traders will understand completely what will be returned to them.

Thanks !

Reply:
IMHO perhaps next:
1. just simply **"filled orders"**
2. or combination **"filled and confirmed orders"**

2nd version is more accurate and descriptive IMHO

Reply:
"filled orders" are very similar to "matched orders", and I've said why it's not good use them as is (note: I didn't say they can't be used in a sentence that's clearer). "confirmed" is very confusing. "filled and confirmed" doesn't make sense IMHO.

Can you **reword** the whole sentence? It's not only about choosing a noun or a phrase, it's the whole sentence that should be clearer. If unable to clearly describe it in 6 words or one sentence, please use more words and more sentences.

Reply:
ok, understood, sure. Thanks !